app/components/works/agreement_component.html.erb

```diff
@@ -2,7 +2,7 @@
   <header>Terms of Deposit *</header>
   <div class="row">
     <div class="form-check col-auto">
-      <% if agree_to_terms? %>
+      <% if agree_to_terms? && form.object.model.fetch(:work).depositor.last_work_terms_agreement %>
```

Comment:
Can you move this from the template to the class?

Reply:
I think this can be folded right into the `agree_to_terms` method as I don't see any other callers.

Reply:
Honestly, it was such a chore gettig the dang thing to work, and with it RIGHT BELOW on line 7, I'm gonna say no to this.  You can make a ticket for it if you like.

4 of us tried to figure out how `agree_to_terms?` could be true here and `form.object.model.fetch(:work).depositor.last_work_terms_agreement` could be nil ... but empirically it was so.

I also don't know why this problem was only triggered when the selected embargo release date was an invalid date ... but that was also true.

Reply:
See also #2243.

Reply:
I feel like you are contravening our team norms (https://github.com/sul-dlss/DeveloperPlaybook/blob/main/best-practices/testing.md#why) by not including a meaningful test with this change. This has been a problem for a long time, so let's not rush to push something out that's not ready.